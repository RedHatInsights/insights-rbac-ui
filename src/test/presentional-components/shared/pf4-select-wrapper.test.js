import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FormSelectOption } from '@patternfly/react-core';
import Pf4SelectWrapper from '../../../presentational-components/shared/pf4-select-wrapper';

describe('<Pf4SelectWrapper />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      id: 'bazz',
      input: {
        name: 'bazz',
        value: '',
        onChange: jest.fn(),
      },
      meta: {},
      options: [
        {
          label: 'Foo',
          value: 'bar',
        },
      ],
      FieldProvider: () => <div />,
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should create empty option', () => {
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} />);
    const options = wrapper.find('option');
    expect(options).toHaveLength(2);
  });

  it('should create empty option for required field', () => {
    const wrapper = mount(<Pf4SelectWrapper isRequired {...initialProps} />);
    const options = wrapper.find(FormSelectOption);
    expect(options).toHaveLength(2);
    expect(options.first().props().label).toEqual('Please choose');
  });

  it('should not create empty option', () => {
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} options={[{ label: 'Foo', value: 'bar' }, { label: 'Empty value' }]} />);
    const options = wrapper.find('option');
    expect(options).toHaveLength(2);
  });

  it('should not create empty option if select has value and is required', () => {
    const wrapper = mount(
      <Pf4SelectWrapper
        isRequired
        {...initialProps}
        input={{ ...initialProps.input, value: 'some value' }}
        options={[{ label: 'Foo', value: 'bar' }]}
      />
    );
    const options = wrapper.find('option');
    expect(options).toHaveLength(1);
  });

  it('should render correctly in error state', () => {
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} meta={{ error: 'Error', touched: true }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with description', () => {
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} description="description" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onChange function', () => {
    const onChange = jest.fn();
    const wrapper = mount(<Pf4SelectWrapper {...initialProps} input={{ ...initialProps.input, onChange }} />);
    wrapper.find('select').simulate('change', { target: { value: 'Foo' } });
    expect(onChange).toHaveBeenCalledWith('', expect.objectContaining({ target: { value: 'Foo' } }));
  });
});
