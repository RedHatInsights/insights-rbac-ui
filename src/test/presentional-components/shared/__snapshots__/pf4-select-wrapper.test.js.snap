// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pf4SelectWrapper /> should render correctly 1`] = `
<Pf4SelectWrapper
  FieldProvider={[Function]}
  id="bazz"
  input={
    Object {
      "name": "bazz",
      "onChange": [MockFunction],
      "value": "",
    }
  }
  meta={Object {}}
  options={
    Array [
      Object {
        "label": "Foo",
        "value": "bar",
      },
    ]
  }
>
  <FormGroup
    fieldId="bazz"
    isValid={true}
  >
    <div
      className="pf-c-form__group"
    >
      <Select
        FieldProvider={[Function]}
        id="bazz"
        input={
          Object {
            "name": "bazz",
            "onChange": [MockFunction],
            "value": "",
          }
        }
        isValid={true}
        options={
          Array [
            Object {
              "label": "Foo",
              "value": "bar",
            },
          ]
        }
      >
        <FormSelect
          className=""
          id="bazz"
          isDisabled={false}
          isRequired={false}
          isValid={true}
          name="bazz"
          onBlur={[Function]}
          onChange={[MockFunction]}
          onFocus={[Function]}
          validated="default"
          value=""
        >
          <select
            aria-invalid={false}
            className="pf-c-form-control"
            disabled={false}
            id="bazz"
            name="bazz"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value=""
          >
            <FormSelectOption
              key="None"
              label="None"
            >
              <option
                className=""
                disabled={false}
                value=""
              >
                None
              </option>
            </FormSelectOption>
            <FormSelectOption
              key="bar"
              label="Foo"
              value="bar"
            >
              <option
                className=""
                disabled={false}
                value="bar"
              >
                Foo
              </option>
            </FormSelectOption>
          </select>
        </FormSelect>
      </Select>
    </div>
  </FormGroup>
</Pf4SelectWrapper>
`;

exports[`<Pf4SelectWrapper /> should render correctly in error state 1`] = `
<Pf4SelectWrapper
  FieldProvider={[Function]}
  id="bazz"
  input={
    Object {
      "name": "bazz",
      "onChange": [MockFunction],
      "value": "",
    }
  }
  meta={
    Object {
      "error": "Error",
      "touched": true,
    }
  }
  options={
    Array [
      Object {
        "label": "Foo",
        "value": "bar",
      },
    ]
  }
>
  <FormGroup
    fieldId="bazz"
    helperTextInvalid="Error"
    isValid={false}
  >
    <div
      className="pf-c-form__group"
    >
      <Select
        FieldProvider={[Function]}
        id="bazz"
        input={
          Object {
            "name": "bazz",
            "onChange": [MockFunction],
            "value": "",
          }
        }
        isValid={false}
        options={
          Array [
            Object {
              "label": "Foo",
              "value": "bar",
            },
          ]
        }
      >
        <FormSelect
          className=""
          id="bazz"
          isDisabled={false}
          isRequired={false}
          isValid={false}
          name="bazz"
          onBlur={[Function]}
          onChange={[MockFunction]}
          onFocus={[Function]}
          validated="default"
          value=""
        >
          <select
            aria-invalid={true}
            className="pf-c-form-control"
            disabled={false}
            id="bazz"
            name="bazz"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value=""
          >
            <FormSelectOption
              key="None"
              label="None"
            >
              <option
                className=""
                disabled={false}
                value=""
              >
                None
              </option>
            </FormSelectOption>
            <FormSelectOption
              key="bar"
              label="Foo"
              value="bar"
            >
              <option
                className=""
                disabled={false}
                value="bar"
              >
                Foo
              </option>
            </FormSelectOption>
          </select>
        </FormSelect>
      </Select>
      <div
        aria-live="polite"
        className="pf-c-form__helper-text pf-m-error"
        id="bazz-helper"
      >
        Error
      </div>
    </div>
  </FormGroup>
</Pf4SelectWrapper>
`;

exports[`<Pf4SelectWrapper /> should render correctly with description 1`] = `
<Pf4SelectWrapper
  FieldProvider={[Function]}
  description="description"
  id="bazz"
  input={
    Object {
      "name": "bazz",
      "onChange": [MockFunction],
      "value": "",
    }
  }
  meta={Object {}}
  options={
    Array [
      Object {
        "label": "Foo",
        "value": "bar",
      },
    ]
  }
>
  <FormGroup
    fieldId="bazz"
    isValid={true}
  >
    <div
      className="pf-c-form__group"
    >
      <TextContent>
        <div
          className="pf-c-content"
        >
          <Text
            component="small"
          >
            <small
              className=""
              data-pf-content={true}
            >
              description
            </small>
          </Text>
        </div>
      </TextContent>
      <Select
        FieldProvider={[Function]}
        id="bazz"
        input={
          Object {
            "name": "bazz",
            "onChange": [MockFunction],
            "value": "",
          }
        }
        isValid={true}
        options={
          Array [
            Object {
              "label": "Foo",
              "value": "bar",
            },
          ]
        }
      >
        <FormSelect
          className=""
          id="bazz"
          isDisabled={false}
          isRequired={false}
          isValid={true}
          name="bazz"
          onBlur={[Function]}
          onChange={[MockFunction]}
          onFocus={[Function]}
          validated="default"
          value=""
        >
          <select
            aria-invalid={false}
            className="pf-c-form-control"
            disabled={false}
            id="bazz"
            name="bazz"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            value=""
          >
            <FormSelectOption
              key="None"
              label="None"
            >
              <option
                className=""
                disabled={false}
                value=""
              >
                None
              </option>
            </FormSelectOption>
            <FormSelectOption
              key="bar"
              label="Foo"
              value="bar"
            >
              <option
                className=""
                disabled={false}
                value="bar"
              >
                Foo
              </option>
            </FormSelectOption>
          </select>
        </FormSelect>
      </Select>
    </div>
  </FormGroup>
</Pf4SelectWrapper>
`;
